src/CoreShop/Component/Resource/Reflection/ClassReflection.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * CoreShop
+ *
+ * This source file is available under two different licenses:
+ *  - GNU General Public License version 3 (GPLv3)
+ *  - CoreShop Commercial License (CCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @copyright  Copyright (c) CoreShop GmbH (https://www.coreshop.org)
+ * @license    https://www.coreshop.org/license     GPLv3 and CCL
+ *
+ */
+
+namespace CoreShop\Component\Resource\Reflection;
+
+use Symfony\Component\Finder\Finder;
+
+final class ClassReflection
+{
+    public static function getResourcesByPaths(array $paths): iterable
+    {
+        foreach ($paths as $resourceDirectory) {
+            $resources = self::getResourcesByPath($resourceDirectory);
+
+            foreach ($resources as $className) {
+                yield $className;
+            }
```

Comment:
Can't you do it like this?
```suggestion
            yield from self::getResourcesByPath($resourceDirectory);
```